pkg/envinfo/storage.go

```diff
@@ -55,16 +55,12 @@ func (ei *EnvInfo) GetStorage(name string) (*localConfigProvider.LocalStorage, e
 
 // CreateStorage sets the storage related information in the local configuration
 func (ei *EnvInfo) CreateStorage(storage localConfigProvider.LocalStorage) error {
-	err := ei.devfileObj.Data.AddVolume(devfilev1.Component{
-		Name: storage.Name,
-		ComponentUnion: devfilev1.ComponentUnion{
-			Volume: &devfilev1.VolumeComponent{
-				Volume: devfilev1.Volume{
-					Size: storage.Size,
-				},
-			},
+	err := ei.devfileObj.Data.AddVolumeMounts(ei.componentSettings.Name, []devfilev1.VolumeMount{
```

Comment:
this new func adds only volume mounts to the container component in the devfile.

you need to make a new call to add the volume component which was inherently done by the prev function. So just need to call `AddComponents(components []v1.Component) error`

Reply:
Addressed in c4bc9d033. PTAL.

Reply:
looks good

Reply:
@dharmit sorry, I've unresolved this today. Just came back to realize that this has the same problem as https://github.com/openshift/odo/pull/4465#discussion_r596964320

devfile component name != odo component name, a devfile container is a component in devfile schema

Reply:
Ack. PTAL https://github.com/openshift/odo/pull/4465/commits/5b494c927dfecfd5fa0bea683436356576d1d503.

Reply:
looks good!